staging/src/k8s.io/apiserver/pkg/audit/policy/reader.go

```diff
@@ -40,11 +45,25 @@ func LoadPolicyFromFile(filePath string) (*auditinternal.Policy, error) {
 	}
 
 	policy := &auditinternal.Policy{}
-	decoder := audit.Codecs.UniversalDecoder(auditv1beta1.SchemeGroupVersion, auditv1alpha1.SchemeGroupVersion)
-	if err := runtime.DecodeInto(decoder, policyDef, policy); err != nil {
+	decoder := audit.Codecs.UniversalDecoder(apiGroupVersions...)
+
+	_, gvk, err := decoder.Decode(policyDef, nil, policy)
+	if err != nil {
 		return nil, fmt.Errorf("failed decoding file %q: %v", filePath, err)
 	}
 
+	// Ensure the policy file contained an apiVersion and kind.
+	found := false
+	for _, gv := range apiGroupVersions {
+		if gv.Version == gvk.Version && gv.Group == gvk.Group {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return nil, fmt.Errorf("policy file contained invalid apiVersion and kind field combination")
```

Comment:
this "combination" wording sounds strange. "unknown group version %v in policy file %q"